python/proton/_message.py

```diff
@@ -515,20 +515,20 @@ def _set_annotations(self, annotations):
 
     def encode(self):
         self._pre_encode()
-        bytes = ffi.new("char *")
         sz = 16
-        size = ffi.new("size_t *", sz)
         while True:
-            data = pn_message_encode(self._msg, bytes, size)
-            if data == PN_OVERFLOW:
+            size = ffi.new('size_t *', sz)
+            bytes = ffi.new("char []", sz)
+            status_code = pn_message_encode(self._msg, bytes, size)
+            if status_code == PN_OVERFLOW:
                 sz *= 2
                 continue
             else:
-                self._check(err)
-                return data
+                self._check(status_code)
+                return (bytes, size)
```

Comment:
Is this necessary? Can't you only return `bytes`, and then get `size` by calling `len(bytes)`? Wouldn't that meant less changes are necessary in the rest of the code?